### PR TITLE
Refactor and add tests for dashboard params

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Refactors the Dashboard hack so it is only applied to deferred PaymentIntents
- Adds unit tests

## Motivation
https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2650
https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2653

## Testing
- New unit tests

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift [hunk 1]
```diff
@@ -17,6 +17,8 @@ extension PaymentSheet {
         authenticationContext: STPAuthenticationContext,
         paymentHandler: STPPaymentHandler,
         isFlowController: Bool,
+        isDashboardApp: Bool,
+        dashboardParamsUpdater: @escaping DashboardParamsUpdater = setParamsForDashboardApp(confirmType:paymentIntentParams:paymentIntent:configuration:),
```
Comment (eurias-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift:21]: Will this ever be anything other than the default value? If we don't have a case where we need to pass in a different updater here, I would just remove it and call the default function directly

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift [hunk 2]
```diff
@@ -17,6 +17,8 @@ extension PaymentSheet {
         authenticationContext: STPAuthenticationContext,
         paymentHandler: STPPaymentHandler,
         isFlowController: Bool,
+        isDashboardApp: Bool,
```
Comment (eurias-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift:20]: Do we need this param? Can't we just check the key here directly?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift (hunk 1 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift:21]: > Will this ever be anything other than the default value?

Nope! The only reason I did this was for testing so we can test that the `dashboardParamsUpdater` is called under the correct circumstances. https://github.com/stripe/stripe-ios/pull/2984/files#diff-d596d94b7af6c551f33f81a8feceef006d2345d8af7718e26fff50f52e22cbbeR108

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift (hunk 2 above)
Comment (eurias-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift:20]: I'm not sure how I feel about adding extra params (and thus extra complexity) just for the sake of testing, plus this makes the code more tightly bound to dashboard.
Also, this param could potentially be out of sync with the key (we could use a uk and set param to false, or viceversa).

**Review by eurias-stripe — APPROVED** [review]

**Review by eurias-stripe — APPROVED** [review]

**Pull request merged**